gremlin-go/driver/serializer.go

```diff
@@ -165,135 +173,147 @@ func uuidToBigInt(requestID uuid.UUID) big.Int {
+	msg.responseResult.data, err = readFullyQualifiedNullable(&message, &i, true)
 	if err != nil {
 		return msg, err
 	}
+	return msg, nil
+}
 
-	msg.responseID = msgUUID
-	msg.responseStatus.code = uint16(statusCode)
-	msg.responseStatus.message = statusMessage
-	msg.responseStatus.attributes = statusAttributes
-	msg.responseResult.meta = metaAttributes
-	msg.responseResult.data = data
+func initSerializers() {
+	// todo: lock or don't care?
+	// todo: lazy init or don't care?
+	if serializers == nil || len(serializers) == 0 {
+		serializers = map[dataType]writer{
+			bytecodeType:   bytecodeWriter,
+			stringType:     stringWriter,
+			bigIntegerType: bigIntWriter,
+			longType:       longWriter,
+			intType:        intWriter,
+			shortType:      shortWriter,
+			byteType: func(value interface{}, buffer *bytes.Buffer, typeSerializer *graphBinaryTypeSerializer) ([]byte, error) {
+				err := binary.Write(buffer, binary.BigEndian, value.(uint8))
+				return buffer.Bytes(), err
+			},
+			booleanType: func(value interface{}, buffer *bytes.Buffer, typeSerializer *graphBinaryTypeSerializer) ([]byte, error) {
+				err := binary.Write(buffer, binary.BigEndian, value.(bool))
+				return buffer.Bytes(), err
+			},
+			uuidType: func(value interface{}, buffer *bytes.Buffer, typeSerializer *graphBinaryTypeSerializer) ([]byte, error) {
+				err := binary.Write(buffer, binary.BigEndian, value)
+				return buffer.Bytes(), err
+			},
+			floatType: func(value interface{}, buffer *bytes.Buffer, typeSerializer *graphBinaryTypeSerializer) ([]byte, error) {
+				err := binary.Write(buffer, binary.BigEndian, value)
+				return buffer.Bytes(), err
+			},
+			doubleType: func(value interface{}, buffer *bytes.Buffer, typeSerializer *graphBinaryTypeSerializer) ([]byte, error) {
+				err := binary.Write(buffer, binary.BigEndian, value)
+				return buffer.Bytes(), err
+			},
+			vertexType:            vertexWriter,
+			edgeType:              edgeWriter,
+			propertyType:          propertyWriter,
+			vertexPropertyType:    vertexPropertyWriter,
+			lambdaType:            lambdaWriter,
+			traversalStrategyType: traversalStrategyWriter,
+			pathType:              pathWriter,
+			setType:               setWriter,
+			dateType:              timeWriter,
+			durationType:          durationWriter,
+			cardinalityType:       enumWriter,
+			columnType:            enumWriter,
+			directionType:         enumWriter,
+			operatorType:          enumWriter,
+			orderType:             enumWriter,
+			pickType:              enumWriter,
+			popType:               enumWriter,
+			tType:                 enumWriter,
+			barrierType:           enumWriter,
+			scopeType:             enumWriter,
+			pType:                 pWriter,
+			textPType:             textPWriter,
+			bindingType:           bindingWriter,
+			mapType:               mapWriter,
+			listType:              listWriter,
+		}
+	}
+}
 
-	return msg, nil
+func initDeserializers() {
+	// todo: lock or don't care?
```

Comment:
Good question. 

I think worst case we have a situation of overwriting the deserializers with an equivalent object, so effectively no change.

I don't see a need to lock.

Reply:
remove comment